pkgs/racket-doc/scribblings/reference/filesystem.scrbl

```diff
@@ -512,6 +512,70 @@ In all modes, the @exnraise[exn:fail:filesystem] on error (e.g., if no
 such file exists).}
 
 
+@defproc[(file-or-directory-stat [path path-string?]
+                                 [as-link? boolean? #f])
+         (hash/c symbol? any/c)]{
+
+@index['("inode")]{Returns} a hash with the following keys and values:
+
+@itemlist[
+ @item{@racketvalfont{'device-id} : device id}
+ @item{@racketvalfont{'inode} : inode number}
+ @item{@racketvalfont{'mode} : mode bits (see below)}
+ @;{
+ @item{@racketvalfont{'type} : one of @racketvalfont{'socket},
+   @racketvalfont{'symbolic-link}, @racketvalfont{'file}, @racketvalfont{'directory},
+   @racketvalfont{'block-device}, @racketvalfont{'character-device} or
+   @racketvalfont{'fifo}}
+ }
+ @item{@racketvalfont{'hardlink-count} : number of hard links}
+ @item{@racketvalfont{'user-id} : user id of owner}
+ @item{@racketvalfont{'group-id} : group id of owner}
+ @item{@racketvalfont{'device-id-for-special-file} : device id (if special file)}
+ @item{@racketvalfont{'size} : size of file or symbolic link in bytes}
+ @item{@racketvalfont{'block-size} : size of filesystem blocks}
+ @item{@racketvalfont{'block-count} : number of used filesystem blocks}
+ @item{@racketvalfont{'access-time-seconds} : last access time in seconds since the epoch}
+ @item{@racketvalfont{'modify-time-seconds} : last modification time in
+   seconds since the epoch}
+ @item{@racketvalfont{'change-time-seconds} : last status change time in
+   seconds since the epoch}
+ @item{@racketvalfont{'creation-time-seconds} : creation time in seconds since
+   the epoch}
+ @item{@racketvalfont{'access-time-nanoseconds} : last access time in
+   nanoseconds since the epoch}
+ @item{@racketvalfont{'modify-time-nanoseconds} : last modification time in
+   nanoseconds since the epoch}
+ @item{@racketvalfont{'change-time-nanoseconds} : last status change time in
+   nanoseconds since the epoch}
+ @item{@racketvalfont{'creation-time-nanoseconds} : creation time in
+   nanoseconds since the epoch}
+]
+
+If @racket[as-link?] is a true value, then if @racket[path] refers to a
+symbolic link, the stat information of the link is returned instead of the stat
+information of the referenced filesystem item.
+
+The mode bits are the bits for permissions and other data, as returned from the
+Posix @tt{stat}/@tt{lstat} functions or the Windows @tt{_wstat64} function,
+respectively. To select portions of the bit pattern, use the constants
+@racket[user-read-bit] etc.
+
+Depending on the operating system and filesystem, the "nanoseconds" timestamps
```

Comment:
nit: should be
```
``nanoseconds''
```
instead of
```
"nanoseconds"
```

Reply:
Great, I didn't know that/how Scribble supported this. :-) Changed in sschwarzer@bf7f3053 .